### ISSUE TITLE
inherited: command not found

### ISSUE THREAD

**Issue by lkpHub** [flutter/flutter#73131]
I am receiving the below error when running a build for a flutter app on IOS with a physical device on a mac M1 chip set. 
Other people can run normally, only I have the following error, I don’t know if it is caused by the M1 chip, the others are the i5 chip

Version Information:
Xcode: 12.3 MacOs: 11.0.1 flutter: 1.22.4 Android Studio: 4.1.1

Error message:

```
    export YACC\=yacc
    export arch\=undefined_arch
    export variant\=normal
    /bin/sh -c /Users/lkp_mac/Library/Developer/Xcode/DerivedData/Yonline-dseboznqvnxvyoezrmzdgtvhsqke/Build/Intermediates.noindex/Yonline.build/Debug-iphonesimulator/Yonline.build/Script-B3D5927362BB91FAB37F2DB6.sh

/Users/lkp_mac/Desktop/Carzone/Yonline/../flutter_yonline/.ios/Flutter/flutter_export_environment.sh: line 8: inherited: command not found

Command PhaseScriptExecution failed with a nonzero exit code
```

This problem has troubled me for a long time, and it caused me to be unable to run the project normally! Is it caused by the M1 chip on my side? Would you like to change your computer? Please help me

**Comment by lkpHub** [comment]
Has anyone else encountered this problem?

**Comment by TahaTesser** [comment]
Hi @lkpHub 
Can you please provide your `flutter doctor -v`, your `flutter run --verbose > flutter_run.log` logs (please provide this log file), and a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example)?
Thank you

**Comment by lkpHub** [comment]
flutter doctor -v. mas:
```
[✓] Flutter (Channel stable, 1.22.4, on macOS 11.0.1 20B29 darwin-arm, locale
    zh-Hans-CN)
    • Flutter version 1.22.4 at /Users/lkp_mac/Desktop/FlutterSDK/flutter
    • Framework revision 1aafb3a8b9 (7 weeks ago), 2020-11-13 09:59:28 -0800
    • Engine revision 2c956a31c0
    • Dart version 2.10.4
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/lkp_mac/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.9.1

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin installed
    • Dart plugin version 201.9317
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] Connected device (1 available)
    • iPhone 12 Pro Max (mobile) • 3E9219F3-2D8A-48EA-B834-D73A5A869B63 • ios •
      com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)

• No issues found!
```

log msg:
```
[  +32 ms] Error: No pubspec.yaml file found.
           This command should be run from the root of your Flutter project.
           Do not run this command from the root of your git clone of Flutter.
[   +5 ms] 
           #0      FlutterCommand.validateCommand (package:flutter_tools/src/runner/flutter_command.dart:1074:11)
           #1      RunCommand.validateCommand (package:flutter_tools/src/commands/run.dart:348:19)
           #2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:933:11)
           <asynchronous suspension>
           #3      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:836:33)
           <asynchronous suspension>
           #4      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart)
           #5      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:29)
           #6      _rootRun (dart:async/zone.dart:1190:13)
           #7      _CustomZone.run (dart:async/zone.dart:1093:19)
           #8      _runZoned (dart:async/zone.dart:1630:10)
           #9      runZoned (dart:async/zone.dart:1550:10)
           #10     AppContext.run (package:flutter_tools/src/base/context.dart:149:18)
           #11     FlutterCommand.run (package:flutter_tools/src/runner/flutter_command.dart:825:20)
           #12     CommandRunner.runCommand (package:args/command_runner.dart:197:27)
           #13     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:335:21)
           #14     _rootRunUnary (dart:async/zone.dart:1198:47)
           #15     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #16     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #17     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #18     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #19     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #20     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
           #21     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
           #22     FlutterVersion.checkFlutterVersionFreshness (package:flutter_tools/src/version.dart)
           #23     _rootRunUnary (dart:async/zone.dart:1198:47)
           #24     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #25     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #26     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #27     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #28     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #29     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
           #30     _rootRun (dart:async/zone.dart:1190:13)
           #31     _CustomZone.run (dart:async/zone.dart:1093:19)
           #32     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
           #33     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
           #34     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
           #35     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
           #36     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
           #37     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
```

**Comment by lkpHub** [comment]
@TahaTesser The above comment is log information

**Comment by GaspardMerten** [comment]
Well you don't seem to be quite in the right folder! It seems that you are running the build command inside of flutter's own folder. You can try to use the terminal, cd into your project root and then run flutter build / run.

**Comment by lkpHub** [comment]
The above problem still occurs. I mixed flutter with Xcode. This error message is the error message given when Xcode is run.

**Comment by GaspardMerten** [comment]
What do you mean by mixing Flutter ans XCode?

What I would try would be:

```

cd my project path 
flutter clean
flutter pub get
cd ios 
pod install
```

Open the Xcode workspace file which is inside your project root folder. Run XCODE build / run command from the GUI.

**Comment by lkpHub** [comment]
@GaspardMerten Thank you so much for your help! This problem is my own M1 problem. When I used Xcode to compile, I did not choose to use Rosetta, which caused the problem! When you choose to use Rosetta to open Xcode, there is no problem!

The first error message of this question is
```
building for iOS Simulator, but linking in object file built for iOS, for architecture arm64”
```
I modified the configuration information, which resulted in the above error message! The problem has been solved, thank you very much for your help
